torch/_inductor/codegen/common.py

```diff
@@ -57,10 +57,15 @@ def _print_Pow(self, expr):
         if exp > 0:
             return "*".join([self.paren(base)] * exp)
         elif exp < 0:
-            return "1/" + self.paren("*".join([self.paren(base)] * abs(exp)))
+            return "1.0/" + self.paren("*".join([self.paren(base)] * abs(exp)))
```

Comment:
I guess maybe I can accept this, because in Python this seems semantics preserving? But if ExprPrinter is printing Python expressions, as it seems to be doing here, then it seems to me that this is still not getting at the root problem, which is that whatever is ingesting divisions in the cpu C++ backend isn't interpreting them as truediv. Can you dig deeper here?

Reply:
It appears to me the truediv in sympy is always represented as pow with `-1` exponent. So, this change would be good enough to solve the problem of c++ backend?

Reply:
@ezyang `CppPrinter` inherits from `ExprPrinter`, so this string ends up in C++ code. (It's not eval'd immediately because of symbolic args.) In Python, `/` is always truediv, while intdiv is a different op (`//`). In C++, it depends on arg types: for two int args, the result will be an int. With a float constant, the code will always use truediv (producing the same results in C++ and Python).

SymPy also uses float constants in the C printer to avoid this problem. From `sympy/sympy/printing/c.py`:

```py
    @_as_macro_if_defined
    def _print_Pow(self, expr):
        if "Pow" in self.known_functions:
            return self._print_Function(expr)
        PREC = precedence(expr)
        suffix = self._get_func_suffix(real)
        if equal_valued(expr.exp, -1):
            literal_suffix = self._get_literal_suffix(real)
            return '1.0%s/%s' % (literal_suffix, self.parenthesize(expr.base, PREC))
        elif equal_valued(expr.exp, 0.5):
            return '%ssqrt%s(%s)' % (self._ns, suffix, self._print(expr.base))
        elif expr.exp == S.One/3 and self.standard != 'C89':
            return '%scbrt%s(%s)' % (self._ns, suffix, self._print(expr.base))
        else:
            return '%spow%s(%s, %s)' % (self._ns, suffix, self._print(expr.base),
                                   self._print(expr.exp))

    def _print_Rational(self, expr):
        p, q = int(expr.p), int(expr.q)
        suffix = self._get_literal_suffix(real)
        return '%d.0%s/%d.0%s' % (p, suffix, q, suffix)
``` 

So I think this is fine as is, please elaborate if I'm missing something.

Reply:
Should this go to CppPrinter then, and not to common?

Reply:
> Should this go to CppPrinter then, and not to common?

It could, but I didn't want to duplicate 99% of this method just to add `.0` (while it works for C++ and Python). Code duplication could end up being problematic later if impls diverge, so I avoid it if possible. LMK if you still think that moving this to C++ is better.